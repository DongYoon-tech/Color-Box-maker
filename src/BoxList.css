.BoxList{
    text-align: center;
    padding: 20px;

}

