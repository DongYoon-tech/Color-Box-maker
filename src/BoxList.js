import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import "./BoxList.css";

const BoxList = () => {

    const [boxes, setBoxes] = useState([])

    const FormSubmit = (data) => {
        setBoxes(c => [...c, data])
    }

    const removeBox = (id) => {
        setBoxes(boxes => boxes.filter(box => box.id !== id));
    }

    const renderBox = () => {
        return (
            <div className="BoxList-box">
                {boxes.map(b => (
                    <Box
                        id={b.id}
                        color={b.backgroundColor}
                        w={b.width}
                        h={b.height}
                        removeBox={removeBox} />
                ))}
            </div>
        )
    }

    return (
        <div className="BoxList">
            <NewBoxForm FormSubmit={FormSubmit} />
            {renderBox()}
        </div>
    )
}

export default BoxList;