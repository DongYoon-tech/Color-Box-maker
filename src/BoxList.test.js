import React from "react";
import { fireEvent, render } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function () {
    render(<BoxList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

it("add a new box", function () {
    const { getByLabelText, queryByText, getAllByDisplayValue, getByText } = render(<BoxList />);

    expect(queryByText('x')).not.toBeInTheDocument();

    const boxColor = getByLabelText("Color");
    const boxHeight = getByLabelText("Height");
    const boxWidth = getByLabelText("Width");
    const Addbtn = queryByText("Add Box");

    fireEvent.change(boxColor, { target: { value: "red" } });
    fireEvent.change(boxHeight, { target: { value: "100" } });
    fireEvent.change(boxWidth, { target: { value: "100" } });
    fireEvent.click(Addbtn)

    const removeBtn = getByText('x');
    expect(removeBtn).toBeInTheDocument();
    expect(removeBtn.previousSibling).toHaveStyle(`
    width: 100px;
    height: 100px;
    background-color: red;
  `);
    expect(getAllByDisplayValue("")).toHaveLength(3);
});

it("remove a box", function () {
    const { getByLabelText, queryByText, getByText } = render(<BoxList />);

    const boxColor = getByLabelText("Color");
    const boxHeight = getByLabelText("Height");
    const boxWidth = getByLabelText("Width");
    const Addbtn = queryByText("Add Box");

    fireEvent.change(boxColor, { target: { value: "red" } });
    fireEvent.change(boxHeight, { target: { value: "100" } });
    fireEvent.change(boxWidth, { target: { value: "100" } });
    fireEvent.click(Addbtn)

    const removeBtn = getByText('x');

    fireEvent.click(removeBtn);
    expect(removeBtn).not.toBeInTheDocument();
});