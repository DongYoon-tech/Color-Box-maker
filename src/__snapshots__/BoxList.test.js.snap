// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <div>
      <form
        class="NewBoxForm"
      >
        <div>
          <label
            for="backgroundColor"
          >
            Color
          </label>
          <input
            id="backgroundColor"
            name="backgroundColor"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="width"
          >
            Width
          </label>
          <input
            id="width"
            name="width"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="height"
          >
            Height
          </label>
          <input
            id="height"
            name="height"
            type="text"
            value=""
          />
        </div>
        <button>
          Add Box
        </button>
      </form>
    </div>
    <div
      class="BoxList-box"
    />
  </div>
</DocumentFragment>
`;
